step-cli/step-cli-core/src/main/java/step/cli/parameters/ApParameters.java

```diff
@@ -0,0 +1,77 @@
+package step.cli.parameters;
+
+import step.cli.StepCliExecutionException;
+import step.core.maven.MavenArtifactIdentifier;
+
+import java.io.File;
+
+public abstract class ApParameters<T extends ApParameters<T>> extends Parameters<T> {
+
+    private File automationPackageFile;
+    private MavenArtifactIdentifier automationPackageMavenArtifact;
+    private MavenArtifactIdentifier automationPackageLibraryMavenArtifact;
+    private File automationPackageLibraryFile;
+    private String automationPackageManagedLibraryName;
+
+    /**
+     * @return the maven snippet of automation package to be uploaded or null to use the local file instead.
+     */
+    public MavenArtifactIdentifier getAutomationPackageMavenArtifact() {
+        return automationPackageMavenArtifact;
+    }
+
+    public T setAutomationPackageMavenArtifact(MavenArtifactIdentifier automationPackageMavenArtifact) {
+        this.automationPackageMavenArtifact = automationPackageMavenArtifact;
+        //noinspection unchecked
```

Comment:
Alternative to repeating the noinspection for each relevant statement: `@SuppressWarnings("unchecked")` on the entire class. Since this class is basically trivial, I personally have a minimal preference for the class-level one, but both are fine with me.